### PR TITLE
Fix cpplint test in motion_computation to support CI/CD

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
Fixing motion_computation cpplint errors introduced in here:
https://github.com/usdot-fhwa-stol/carma-platform/pull/2399
with the failing github action:
https://github.com/usdot-fhwa-stol/carma-platform/actions/runs/9449278789/job/26037602316?pr=2397#step:13:23837

My environment was never configured for clang format and only used to delete whitespaces. I improved it by:
1. I have installed clang by `sudo apt install clang`
2. installed cpplint by `sudo pip install cpplint`
3. I have then installed vscode extensions that use packages Clang-Format and cpplint
4. created a .clang-format file outside the src folder that has all the repos and started editing to get the format. If there is multiple, I believe the immediate .clang-format file in the project gets to be used first.
5. added following to my user settings.json in vscode (F1 -> "Preferences: Open user settings.json")
```
    "[cpp]": {
        "editor.defaultFormatter": "xaver.clang-format"
    },
    "editor.wordWrap": "wordWrapColumn",
    "editor.wrappingIndent": "indent",
    "editor.wordWrapColumn": 100,
    "editor.formatOnSave": true,
    "editor.rulers" :[
        100
    ],
    "C_Cpp.clang_format_fallbackStyle": "Google",
    "cpplint.lineLength": 100
```
For a lot of the things to take effect and add vertical line indicating the 100 line limit, which one can make false

The cpplint in the CICD is still complaining about the order of the headers. After trying multiple orders, I found an [order in autoware.auto](https://github.com/usdot-fhwa-stol/autoware.auto/blob/carma-system-4.5.0/AutowareAuto/.clang-format) that does the job. Despite visually it is correct, not sure why cpplint is still complaining (which `carma_cooperative_perception` also seems to have encountered). Therefore, turning off the check for include orders for now as we haven't established strict convention with it anyways.

## Related GitHub Issue
NA

## Related Jira Key
NA

## Motivation and Context
NA

## How Has This Been Tested?
CICD passes

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: motion_computation/include/motion_computation/motion_computation_worker.hpp [hunk 1]
```diff
@@ -149,7 +149,7 @@ class MotionComputationWorker
   bool enable_ctrv_for_small_vehicle_obj_ = true;
   bool enable_ctrv_for_large_vehicle_obj_ = true;
   bool enable_ctrv_for_pedestrian_obj_ = false;
-  
+
```
Comment (JonSmet) [inline motion_computation/include/motion_computation/motion_computation_worker.hpp:152]: Not a comment on this line specifically, but it looks like the 'Run Tests' CI process is still failing on the motion_computation package for the cpplint tests. Notable lines from the logs included below:

```
...
2024-06-11T18:18:02.7797663Z 3: /opt/carma/src/carma-platform/motion_computation/include/motion_computation/motion_computation_config.hpp:85:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7800243Z 3: /opt/carma/src/carma-platform/motion_computation/include/motion_computation/motion_computation_config.hpp:86:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7802714Z 3: /opt/carma/src/carma-platform/motion_computation/include/motion_computation/motion_computation_worker.hpp:77:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7805083Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_node.cpp:56:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7807507Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_node.cpp:58:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7810201Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_node.cpp:60:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7812786Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_node.cpp:62:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7815576Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_node.cpp:63:  Redundant blank line at the end of a code block should be deleted.  [whitespace/blank_line] [3]
2024-06-11T18:18:02.7818227Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_node.cpp:107:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7820619Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_node.cpp:134:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7822717Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_node.cpp:135:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7825324Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_worker.cpp:140:  Redundant blank line at the end of a code block should be deleted.  [whitespace/blank_line] [3]
2024-06-11T18:18:02.7827986Z 3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_worker.cpp:195:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
2024-06-11T18:18:02.7830408Z 3: Category 'whitespace/blank_line' errors found: 2
2024-06-11T18:18:02.7831271Z 3: Category 'whitespace/line_length' errors found: 11
2024-06-11T18:18:02.7831935Z 3: Total errors found: 13
...
2024-06-11T18:18:02.7924589Z 3: -- run_test.py: return code 1
2024-06-11T18:18:02.7926565Z 3: -- run_test.py: verify result file '/opt/carma/build/motion_computation/test_results/motion_computation/cpplint.xunit.xml'
2024-06-11T18:18:02.8007503Z 3/7 Test #3: cpplint ..........................***Failed    1.04 sec
```

**Review by JonSmet — CHANGES_REQUESTED** [review]

**Comment by MishkaMN** [comment]
Looks like `motion_computation` doesn't have cpp lint failures anymore. But `approaching_emergency_vehicle_plugin` and `gnss_to_map_convertor` have intermittent test failures due to timeouts. Possibly related to: https://github.com/usdot-fhwa-stol/carma-platform/issues/2335
Github Action to track
https://github.com/usdot-fhwa-stol/carma-platform/actions/runs/9490590550/job/26160807595?pr=2405

File: motion_computation/.clang-format [hunk 2]
```diff
@@ -0,0 +1,46 @@
+---
+# ROS 2 C++ style guidelines
```
Comment (JonSmet) [inline motion_computation/.clang-format:2]: Is there a reason we are using a different `.clang-format` here than in some of our other packages (i.e. `carma_cooperative_perception`, `roadway_objects`, etc.?

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: motion_computation/.clang-format (hunk 2 above)
Comment (MishkaMN) [reply motion_computation/.clang-format:2]: This one is mostly same with `carma_cooperative_perception` except `SpacesBeforeTrailingComments: 2` and `ordering of the include files`. `SpacesBeforeTrailingComments: 2` was being complained in the cpplint in one of the runs:
image

https://github.com/usdot-fhwa-stol/carma-platform/actions/runs/9486323826/job/26140411232

You can also see it is complaining about the orders of the include headers. I tried multiple times and think I achieved with the order here I got from autoware. but it was still complaining so I added extra file to disable that check just like `carma_cooperative_perception`

**Review by JonSmet — APPROVED** [review]

**Pull request merged**